vmm/src/vm.rs

```diff
@@ -1206,6 +1218,73 @@ impl AsRawFd for EpollContext {
     }
 }
 
+lazy_static! {
+    pub static ref CVAR: Condvar = Condvar::new();
+    pub static ref MUTEX: Mutex<()> = Mutex::new(());
+}
+pub static MASK: AtomicUsize = AtomicUsize::new(0);
+
+#[cfg(unix)]
+pub extern "C" fn handler(sig: c_int, _: *mut siginfo_t, _: *mut c_void) {
+    let mask = match sig {
+        SIGHUP => 1,
+        SIGINT => 2,
+        SIGWINCH => 4,
+        _ => return,
+    };
+
+    loop {
+        let prev_mask = MASK.load(Ordering::Relaxed);
+        let new_mask = prev_mask | mask;
+        if MASK.compare_and_swap(prev_mask, new_mask, Ordering::Relaxed) == new_mask {
+            break;
+        }
+    }
+    CVAR.notify_all();
+}
+
+pub unsafe fn set_os_handler(sig: i32) {
+    register_signal_handler(
+        sig,
+        vmm_sys_util::signal::SignalHandler::Siginfo(handler),
+        false,
+        SA_SIGINFO,
+    )
+    .expect("Failed to register OS  signal handler");
+}
+
+pub fn set_handler<F>(signals: &[i32], user_handler: F)
+where
+    F: Fn(&[i32]) + 'static + Send,
+{
+    for &signal in signals.iter() {
+        unsafe { set_os_handler(signal) }
+    }
+    thread::spawn(move || {
+        let mut signals = Vec::new();
+        loop {
+            let mask = MASK.load(Ordering::Relaxed);
+            if mask == 0 {
+                let _ = CVAR.wait(MUTEX.lock().unwrap());
+                thread::yield_now();
+                continue;
+            }
+            signals.clear();
+            if mask & 1 != 0 {
+                signals.push(libc::SIGHUP)
```

Comment:
Please remove all `SIGHUP` and `SIGINT` related code since we're not handling it eventually. And also because if we wanted to handle those signals at some point, it should be done in a separate commit/PR.

Reply:
Done.